src/modules/recipies/RecipeForm/index.jsx

```diff
@@ -0,0 +1,69 @@
+import React from "react";
+import { useForm } from "react-hook-form";
+import { useRecipe } from "../RecipeContext";
+import { useNavigate } from "react-router-dom";
+import "./styles.css";
+
+export const RecipeForm = () => {
+    const { addRecipe } = useRecipe();
+    const { register, handleSubmit, reset } = useForm();
+    const navigate = useNavigate();
+
+
+    const onSubmit = (data) => {
+        const formattedRecipe = {
+            idMeal: new Date().getTime().toString(),
+            strMeal: data.strMeal,
+            strInstructions: data.strInstructions,
+            strMealThumb: data.strMealThumb || "default-image.jpg",
+            strCategory: data.strCategory || "Uncategorized",
+        };
+
+        addRecipe(formattedRecipe);
+        reset();
+        navigate("/recipes");
+    };
+
+    const handleBackButton = () => {
+        navigate(-1);
+    };
+
+    return (
+        <>
+            <button onClick={handleBackButton} className="button back-btn">
+                Back
+            </button>
+            <div className="form-container">
+                <form onSubmit={handleSubmit(onSubmit)}>
```

Comment:
It's important that the user can enter (almost) all data for the recipe that we get from API so that the data is in the same format.

API contains this information:

{
      "idMeal": "52771", // generate, don't add input for it
      "strMeal": "Spicy Arrabiata Penne",
      "strDrinkAlternate": null,
      "strCategory": "Vegetarian", // Use select element so use can select from a list of supported categories (check API documentation to see what categories are supported)
      "strArea": "Italian",  // Use select element so use can select from a list of supported areas (check API documentation to see what areas are supported)
      "strInstructions": "...", // Make sure that there's a vlidation rule for this field (e.g. min length)
      "strMealThumb": "https://www.themealdb.com/images/media/meals/ustsqw1468250014.jpg", // You need to add validation rules for this field. Make sure it supports adding images from internet (e.g. test with images from unsplash)
      "strTags": "Pasta,Curry", // optional field
      "strYoutube": "https://www.youtube.com/watch?v=1IszT_guI08", // Similar to the image but should be an optional field
      "strIngredient1": "penne rigate", // For ingredients, make sure that user can add each ingredient and measure as a pair (e.g. add 2 input elements on the same line. Then, you'll need to convert the data to format compatible with API
      "strIngredient2": "olive oil",
      "strIngredient3": "garlic",
      "strIngredient4": "chopped tomatoes",
      "strIngredient5": "red chilli flakes",
      "strIngredient6": "italian seasoning",
      "strIngredient7": "basil",
      "strIngredient8": "Parmigiano-Reggiano",
      "strIngredient9": "",
      "strIngredient10": "",
      "strIngredient11": "",
      "strIngredient12": "",
      "strIngredient13": "",
      "strIngredient14": "",
      "strIngredient15": "",
      "strIngredient16": null,
      "strIngredient17": null,
      "strIngredient18": null,
      "strIngredient19": null,
      "strIngredient20": null,
      "strMeasure1": "1 pound",
      "strMeasure2": "1/4 cup",
      "strMeasure3": "3 cloves",
      "strMeasure4": "1 tin ",
      "strMeasure5": "1/2 teaspoon",
      "strMeasure6": "1/2 teaspoon",
      "strMeasure7": "6 leaves",
      "strMeasure8": "sprinkling",
      "strMeasure9": "",
      "strMeasure10": "",
      "strMeasure11": "",
      "strMeasure12": "",
      "strMeasure13": "",
      "strMeasure14": "",
      "strMeasure15": "",
      "strMeasure16": null,
      "strMeasure17": null,
      "strMeasure18": null,
      "strMeasure19": null,
      "strMeasure20": null,
      "strSource": null, // optional field
      "strImageSource": null, // optional field
      "strCreativeCommonsConfirmed": null, // optional checkbox
      "dateModified": null // not a user input, generate this data when recipe is added
    }